Fixes MDLabel Bug

lemousehunter opened the pull request:
Fixes #515 : Bug where `MDLabel.font_style` does not update

### Description of Changes
Changed `OptionProperty` to `ObjectProperty`
Added `check_font_style()` to replace the validation function of `OptionProperty`

### Code in question:
`font_style = ObjectProperty("Body1")` changed to  `font_style = ObjectProperty("Body1")`
Added:
- `self.check_font_styles()` to `.__init__()`
- ```
    def check_font_styles(self):
        if self.font_style not in list(self.theme_cls.font_styles.keys()):
            raise ValueError(
                "MDLabel.font_style is set to an invalid option '{}'. "
                "Must be one of: {}".format(
                    self.font_style, list(self.theme_cls.font_styles)
                )
            )
```

@ArtemSBulgakov @HeaTTheatR

lemousehunter committed:
Fixes bug

Fixes Bug where MDLabel.font_style does not update

lemousehunter committed:
Fixes bug

Fixes #515 : Bug where MDLabel.font_style does not update
Description of Changes

Changed OptionProperty to ObjectProperty
Added check_font_style function to replace the validation function of OptionProperty

HeaTTheatR reviewed (commented):
Thank you for helping to improve the KivyMD library!
If you write that you have fixed an error, show the code that reproduces the error that you are describing. Also show the code (after applying your changes) that you are using to avoid the error you are describing.

lemousehunter commented:
@HeaTTheatR I just did so

HeaTTheatR commented:
@lemousehunter 
```python
    font_style = ObjectProperty("Body1")                                       # <-------- invalid property
    """
    Label font style.
    Available options are: `'H1'`, `'H2'`, `'H3'`, `'H4'`, `'H5'`, `'H6'`,     # <-------- invalid description
    `'Subtitle1'`, `'Subtitle2'`, `'Body1'`, `'Body2'`, `'Button'`,
    `'Caption'`, `'Overline'`, `'Icon'`.
    :attr:`font_style` is an :class:`~kivy.properties.OptionProperty`.         # <-------- invalid property
    and defaults to `'Body1'`.
    """```

lemousehunter committed:
Fixes MDLabel Font Style Bug

Fixes #515
Bug where MDLabel.font_style does not update

Description of Changes

Changed OptionProperty to ObjectProperty
Added check_font_style() to replace the validation function of OptionProperty
Code in question:

font_style = ObjectProperty("Body1") changed to font_style = ObjectProperty("Body1")
Added:

    self.check_font_styles() to .__init__()

      def check_font_styles(self):
          if self.font_style not in list(self.theme_cls.font_styles.keys()):
              raise ValueError(
                  "MDLabel.font_style is set to an invalid option '{}'. "
                  "Must be one of: {}".format(
                      self.font_style, list(self.theme_cls.font_styles)
                  )
              )
